src/cmdc_tools/datasets/official/MI/data.py

```diff
@@ -0,0 +1,50 @@
+import asyncio
+
+import pandas as pd
+import pyppeteer
+
+from ...base import DatasetBaseNoDate
+from ..base import CountyData
+
+
+class Michigan(DatasetBaseNoDate, CountyData):
+    def get(self):
+        urls = self._get_urls()
+        cases = self._get_cases(urls[0])
+
+        return cases
+
+    def _get_urls(self):
+        urls = asyncio.run(self._get_urls_async())
+        return urls
+
+    async def _get_urls_async(self):
+        browser = await pyppeteer.launch()
+        page = await browser.newPage()
+        await page.goto(
+            "https://www.michigan.gov/coronavirus/0,9753,7-406-98163_98173---,00.html"
+        )
+        # get link to cases
+        cases_url_raw = await page.Jx("//a[text()='Cases by County by Date']/..")
+        cases_url = await cases_url_raw[0].Jeval("a", "node => node.href")
+        # tests_url_raw = await page.Jx("//a[text()='COVID-19 Tests by County']/..")
+        # tests_url = await tests_url_raw[0].Jeval("a", "node => node.href")
+
+        # return [cases_url, tests_url]
+        return [cases_url]
+
+    def _get_cases(self, url):
+        df = pd.read_excel(url)
```

Comment:
Is this successfully parsing `Date` as a `datetime`? It might make sense to add `parse_dates=["Date"]` to the request.

Reply:
It's parsing it. `df.info` shows the `dt` column as a `datetime64`